airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -179,14 +182,20 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
             new RecordMetricInput(connectionUpdaterInput, Optional.of(FailureCause.CANCELED), OssMetricsRegistry.TEMPORAL_WORKFLOW_FAILURE, null));
       }
 
-      if (workflowState.isDeleted()) {
-        if (workflowState.isRunning()) {
-          log.info("Cancelling the current running job because a connection deletion was requested");
-          reportCancelled(connectionUpdaterInput.getConnectionId());
+      final int dontDeleteInTemporal =
+          Workflow.getVersion(DONT_DELETE_IN_TEMPORAL_TAG, Workflow.DEFAULT_VERSION, DONT_DELETE_IN_TEMPORAL_TAG_CURRENT_VERSION);
+
+      if (dontDeleteInTemporal < DONT_DELETE_IN_TEMPORAL_TAG_CURRENT_VERSION) {
+        if (workflowState.isDeleted()) {
```

Comment:
Can combine these two if statements into one one to avoid one layer of nesting.

Reply:
Done